src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,107 @@
-import { useState } from 'react';
+import React, { FC, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { isValidUrl } from '../../helpers/isValidUrl';
+import { Props } from './NewMovie.types';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+export const NewMovie: FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [titleValue, setTitleValue] = useState('');
+  const [descriptionValue, setDescriptionValue] = useState('');
+  const [imgUrlValue, setImgUrlValue] = useState('');
+  const [imdbUrlValue, setImdbUrlValue] = useState('');
+  const [imdbIdValue, setImdbIdValue] = useState('');
+
+  const resetAll = () => {
+    setTitleValue('');
+    setDescriptionValue('');
+    setImgUrlValue('');
+    setImdbUrlValue('');
+    setImdbIdValue('');
+  };
+
+  const isImdbUrlValid = isValidUrl(imgUrlValue);
+  const isImgUrlValid = isValidUrl(imgUrlValue);
+
+  const isSubmitAllowed
+    = titleValue && imgUrlValue && imdbUrlValue
+    && imdbIdValue && isImdbUrlValid && isImgUrlValid;
+
+  const onFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (isSubmitAllowed) {
+      const newMovie: Movie = {
+        title: titleValue,
+        description: descriptionValue,
+        imgUrl: imgUrlValue,
+        imdbUrl: imdbUrlValue,
+        imdbId: imdbIdValue,
+      };
+
+      onAdd(newMovie);
+      setCount(prevCount => prevCount + 1);
+      resetAll();
+    }
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => onFormSubmit(event)}
```

Comment:
```suggestion
      onSubmit={handleFormSubmit}
```

Just pass a reference